python_modules/dagster/dagster/_core/execution/plan/execute_step.py

```diff
@@ -501,9 +509,38 @@ def _get_output_asset_materializations(
     output: Union[Output, DynamicOutput],
     output_def: OutputDefinition,
     io_manager_metadata_entries: Sequence[Union[MetadataEntry, PartitionMetadataEntry]],
+    step_context: StepExecutionContext,
 ) -> Iterator[AssetMaterialization]:
 
     all_metadata = [*output.metadata_entries, *io_manager_metadata_entries]
+
+    # logical version computation
+    asset_layer = step_context.pipeline_def.asset_layer
```

Comment:
This is a fairly large function already - would aid readability to move this into a helper function.

Reply:
good idea